crates/cxx-qt-gen/src/generator/rust/constructor.rs

```diff
@@ -0,0 +1,297 @@
+        let new_cxx = format!("newRs{index}");
+
+        let initialize_rust = format_ident!("initialize_{rust_struct_name_snake}_{index}");
+        let initialize_cxx = format!("initialize{index}");
+
+        let route_arguments_rust =
+            format_ident!("route_arguments_{rust_struct_name_snake}_{index}");
+        let route_arguemnts_cxx = format!("routeArguments{index}");
+
+        let argument_types = &constructor.arguments;
+
+        let route_arguments_parameters: Vec<TokenStream> = constructor
+            .arguments
+            .iter()
+            .enumerate()
+            .map(|(index, ty)| {
+                let name = format_ident!("arg{index}");
+                quote! { #name: #ty }
+            })
+            .collect();
+        let route_arguments_safety = if constructor.arguments.iter().any(types::is_unsafe_cxx_type)
+        {
+            quote! { unsafe }
+        } else {
+            quote! {}
+        };
+
+        let assign_arguments = constructor
+            .arguments
+            .iter()
+            .enumerate()
+            .map(|(index, _ty)| {
+                let name = format_ident!("arg{index}");
+                quote! { #name }
+            })
+            .collect::<Vec<_>>();
+
+        let init_new_arguments = extract_arguments_from_tuple(
+            &constructor.new_arguments,
+            format_ident!("new_arguments"),
+        );
+        let init_initialize_arguments = extract_arguments_from_tuple(
+            &constructor.initialize_arguments,
+            format_ident!("initialize_arguments"),
+        );
+        let init_base_arguments = extract_arguments_from_tuple(
+            &constructor.base_arguments,
+            format_ident!("base_arguments"),
+        );
+
+        let extract_new_arguments = extract_arguments_from_struct(
+            &constructor.new_arguments,
+            format_ident!("new_arguments"),
+        );
+
+        let extract_initialize_arguments = extract_arguments_from_struct(
+            &constructor.initialize_arguments,
+            format_ident!("initialize_arguments"),
+        );
+
+        result.cxx_mod_contents.append(&mut vec![
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #arguments_cxx]
+                #[doc(hidden)]
+                struct #arguments_rust {
+                    baseArguments: #base_arguments_rust,
+                    newArguments: #new_arguments_rust,
+                    initializeArguments: #initialize_arguments_rust,
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #base_arguments_cxx]
+                #[doc(hidden)]
+                struct #base_arguments_rust {
+                    #(#base_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
```

Comment:
unfortunate, assume you can't use phantom data ? And is a bool smaller than a i8 ?

Reply:
Also have you seen upstream if there is an issue for this, could be something to fix if it's simple ?

If there is an issue or you make one, link in a comment

Reply:
I've created an issue in CXX for this.
https://github.com/dtolnay/cxx/issues/1245

This can't be PhantomData I believe, as that wouldn't fix the underlying issue.
In C++ an "empty" struct is actually still 1 byte in size for *reasons*.
In Rust it's actually 0 bytes. PhantomData is 0 bytes as well, so wouldn't change that.
So the memory layout would be different between them without this notEmpty member

An `i8` should be the same size as a `bool` as the minimum addressable size in Rust is one byte.